cores/esp8266/Schedule.cpp

```diff
@@ -80,27 +85,37 @@ bool schedule_function (const std::function<void(void)>& fn)
     return true;
 }
 
-bool schedule_recurrent_function_us (const std::function<bool(void)>& fn, uint32_t repeat_us)
+bool schedule_recurrent_function_us(const std::function<bool(void)>& fn,
+    uint32_t repeat_us, std::function<bool(void)> alarm)
 {
     assert(repeat_us < decltype(recurrent_fn_t::callNow)::neverExpires); //~26800000us (26.8s)
 
-    esp8266::InterruptLock lockAllInterruptsInThisScope;
+    if (!fn)
+        return false;
 
     recurrent_fn_t* item = new (std::nothrow) recurrent_fn_t(repeat_us);
     if (!item)
         return false;
 
     item->mFunc = fn;
+    item->alarm = std::move(alarm);
```

Comment:
I believe `std::move()` is only needed when var is a rvalue reference and useless otherwise. Here alarm is passed by value (not constref, not rvalue-ref)
We use in this file constrefs for all functionals, why not use one for alarm at least for consistency  ?